File: src/Currency.js
const convertCurrency = async () => {
    const inputCurrency = document.getElementById("input_currency_num").value;
    const outputCurrency = document.getElementById("output_currency_num");
    const inputCurrencyType = document.getElementById("input_currency_type").value;
    const outputCurrencyType = document.getElementById("output_currency_type").value;

    switch (inputCurrencyType) {
        case "USD":
            USDCurrency(inputCurrency, outputCurrency, outputCurrencyType);
            break;
        case "CAD":
            CADCurrency(inputCurrency, outputCurrency, outputCurrencyType);
            break;
        case "EUR":
            EURCurrency(inputCurrency, outputCurrency, outputCurrencyType);
            break;
        case "JMD":
            JMDCurrency(inputCurrency, outputCurrency, outputCurrencyType);
            break;
        case "MXN":
            MXNCurrency(inputCurrency, outputCurrency, outputCurrencyType);
            break;
        case "ZAR":
            ZARCurrency(inputCurrency, outputCurrency, outputCurrencyType);
            break;
        case "AUD":
            AUDCurrency(inputCurrency, outputCurrency, outputCurrencyType);
            break;
        case "NZD":
            NZDCurrency(inputCurrency, outputCurrency, outputCurrencyType);
            break;
        case "CHF":
            CHFCurrency(inputCurrency, outputCurrency, outputCurrencyType);
            break;
    }
}

function USDCurrency(inputCurrency, outputCurrency, outputCurrencyType) {
    switch (outputCurrencyType) {
        case "USD":
            convertedCurrency = inputCurrency * 1.0;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "CAD":
            convertedCurrency = inputCurrency * 1.36;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "EUR":
            convertedCurrency = inputCurrency * 0.92;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "JMD":
            convertedCurrency = inputCurrency * 155.24;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "MXN":
            convertedCurrency = inputCurrency * 16.61;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "ZAR":
            convertedCurrency = inputCurrency * 18.18;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "AUD":
            convertedCurrency = inputCurrency * 1.49;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "NZD":
            convertedCurrency = inputCurrency * 1.63;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "CHF":
            convertedCurrency = inputCurrency * 0.91;
            outputCurrency.value = roundNumber(convertedCurrency);
            break; 
    }
}

function CADCurrency(inputCurrency, outputCurrency, outputCurrencyType) {
    switch (outputCurrencyType) {
        case "USD":
            convertedCurrency = inputCurrency * 0.73;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "CAD":
            convertedCurrency = inputCurrency * 1.0;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "EUR":
            convertedCurrency = inputCurrency * 0.68;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "JMD":
            convertedCurrency = inputCurrency * 114.0;
            outputCurrency.value = roundNumber(convertedCurrency);
            break; 
        case "MXN":
            convertedCurrency = inputCurrency * 12.20;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "ZAR":
            convertedCurrency = inputCurrency * 13.35;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "AUD":
            convertedCurrency = inputCurrency * 1.10;
            outputCurrency.value = roundNumber(convertedCurrency);
            break; 
        case "NZD":
            convertedCurrency = inputCurrency * 1.20;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "CHF":
            convertedCurrency = inputCurrency * 0.67;
            outputCurrency.value = roundNumber(convertedCurrency);
            break; 
    }
}

function EURCurrency(inputCurrency, outputCurrency, outputCurrencyType) {
    switch (outputCurrencyType) {
        case "USD":
            convertedCurrency = inputCurrency * 1.09;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "CAD":
            convertedCurrency = inputCurrency * 1.48;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "EUR":
            convertedCurrency = inputCurrency * 1.0;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "JMD":
            convertedCurrency = inputCurrency * 169.06;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "MXN":
            convertedCurrency = inputCurrency * 18.03;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "ZAR":
            convertedCurrency = inputCurrency * 19.76;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "AUD":
            convertedCurrency = inputCurrency * 1.63;
            outputCurrency.value = roundNumber(convertedCurrency);
            break; 
        case "NZD":
            convertedCurrency = inputCurrency * 1.77;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "CHF":
            convertedCurrency = inputCurrency * 0.99;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
    }
}

function JMDCurrency(inputCurrency, outputCurrency, outputCurrencyType) {
    switch (outputCurrencyType) {
        case "USD":
            convertedCurrency = inputCurrency * 0.0064;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "CAD":
            convertedCurrency = inputCurrency * 0.0088;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "EUR":
            convertedCurrency = inputCurrency * 0.0059;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "JMD":
            convertedCurrency = inputCurrency * 1.0;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "MXN":
            convertedCurrency = inputCurrency * 0.11;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "ZAR":
            convertedCurrency = inputCurrency * 0.12;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "AUD":
            convertedCurrency = inputCurrency * 0.0096;
            outputCurrency.value = roundNumber(convertedCurrency);
            break; 
        case "NZD":
            convertedCurrency = inputCurrency * 0.011;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "CHF":
            convertedCurrency = inputCurrency * 0.0059;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
    }
}

function MXNCurrency(inputCurrency, outputCurrency, outputCurrencyType) {
    switch (outputCurrencyType) {
        case "USD":
            convertedCurrency = inputCurrency * 0.060;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "CAD":
            convertedCurrency = inputCurrency * 0.082;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "EUR":
            convertedCurrency = inputCurrency * 0.055;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "JMD":
            convertedCurrency = inputCurrency * 9.35;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "MXN":
            convertedCurrency = inputCurrency * 1.0;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "ZAR":
            convertedCurrency = inputCurrency * 1.09;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "AUD":
            convertedCurrency = inputCurrency * 0.090;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "NZD":
            convertedCurrency = inputCurrency * 0.098;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "CHF":
            convertedCurrency = inputCurrency * 0.055;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
    }
}

function ZARCurrency(inputCurrency, outputCurrency, outputCurrencyType) {
    switch (outputCurrencyType) {
        case "USD":
            convertedCurrency = inputCurrency * 0.055;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "CAD":
            convertedCurrency = inputCurrency * 0.075;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "EUR":
            convertedCurrency = inputCurrency * 0.051;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "JMD":
            convertedCurrency = inputCurrency * 8.54;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "MXN":
            convertedCurrency = inputCurrency * 0.91;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "ZAR":
            convertedCurrency = inputCurrency * 1.0;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "AUD":
            convertedCurrency = inputCurrency * 0.082;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "NZD":
            convertedCurrency = inputCurrency * 0.090;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "CHF":
            convertedCurrency = inputCurrency * 0.050;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
    }
}

function AUDCurrency(inputCurrency, outputCurrency, outputCurrencyType) {
    switch (outputCurrencyType) {
        case "USD":
            convertedCurrency = inputCurrency * 0.67;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "CAD":
            convertedCurrency = inputCurrency * 0.91;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "EUR":
            convertedCurrency = inputCurrency * 0.61;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "JMD":
            convertedCurrency = inputCurrency * 104.03;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "MXN":
            convertedCurrency = inputCurrency * 11.13;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "ZAR":
            convertedCurrency = inputCurrency * 12.17;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "AUD":
            convertedCurrency = inputCurrency * 1.0;
            outputCurrency.value = roundNumber(convertedCurrency);
            break; 
        case "NZD":
            convertedCurrency = inputCurrency * 1.09;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "CHF":
            convertedCurrency = inputCurrency * 0.61;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
    }
}

function NZDCurrency(inputCurrency, outputCurrency, outputCurrencyType) {
    switch (outputCurrencyType) {
        case "USD":
            convertedCurrency = inputCurrency * 0.61;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "CAD":
            convertedCurrency = inputCurrency * 0.83;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "EUR":
            convertedCurrency = inputCurrency * 0.56;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "JMD":
            convertedCurrency = inputCurrency * 95.20;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "MXN":
            convertedCurrency = inputCurrency * 10.18;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "ZAR":
            convertedCurrency = inputCurrency * 11.15;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "AUD":
            convertedCurrency = inputCurrency * 0.92;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "NZD":
            convertedCurrency = inputCurrency * 1.0;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "CHF":
            convertedCurrency = inputCurrency * 0.56;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
    }
}

function CHFCurrency(inputCurrency, outputCurrency, outputCurrencyType) {
    switch (outputCurrencyType) {
        case "USD":
            convertedCurrency = inputCurrency * 1.10;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "CAD":
            convertedCurrency = inputCurrency * 1.50;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "EUR":
            convertedCurrency = inputCurrency * 1.01;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "JMD":
            convertedCurrency = inputCurrency * 170.77;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "MXN":
            convertedCurrency = inputCurrency * 18.27;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "ZAR":
            convertedCurrency = inputCurrency * 20.00;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "AUD":
            convertedCurrency = inputCurrency * 1.64;
            outputCurrency.value = roundNumber(convertedCurrency);
            break; 
        case "NZD":
            convertedCurrency = inputCurrency * 1.79;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
        case "CHF":
            convertedCurrency = inputCurrency * 1.0;
            outputCurrency.value = roundNumber(convertedCurrency);
            break;
    }
}

function roundNumber(num){
    return Number( num.toPrecision(countNonDecimalPlaces(num)+2));
}

function countNonDecimalPlaces(num) {
    let numStr = num.toString();
    let decimalIndex = numStr.indexOf('.');
    return decimalIndex;
  }

window.convertCurrency = convertCurrency;

File: src/Currency.py
def convertCurrency(inputCurrency, outputCurrency, outputCurrencyType):
    match inputCurrency:
        case "USD":
            USDCurrency(inputCurrency, outputCurrency, outputCurrencyType)
        case "CAD":
            CADCurrency(inputCurrency, outputCurrency, outputCurrencyType)
        case "EUR":
            EURCurrency(inputCurrency, outputCurrency, outputCurrencyType)
        case "JMD":
            JMDCurrency(inputCurrency, outputCurrency, outputCurrencyType)
        case "MXN":
            MXNCurrency(inputCurrency, outputCurrency, outputCurrencyType)
        case "ZAR":
            ZARCurrency(inputCurrency, outputCurrency, outputCurrencyType)
        case "AUD":
            AUDCurrency(inputCurrency, outputCurrency, outputCurrencyType)
        case "NZD":
            NZDCurrency(inputCurrency, outputCurrency, outputCurrencyType)
        case "CHF":
            CHFCurrency(inputCurrency, outputCurrency, outputCurrencyType)

def USDCurrency(inputCurrnecy, outputCurrnecy, outputCurrnecyType):
    match outputCurrnecyType:
        case "USD":
            outputCurrnecy.value = inputCurrnecy * 1.0
        case "CAD":
            outputCurrnecy.value = inputCurrnecy * 1.36
        case "EUR":
            outputCurrnecy.value = inputCurrnecy * 0.92
        case "JMD":
            outputCurrnecy.value = inputCurrnecy * 155.24
        case "MXN":
            outputCurrnecy.value = inputCurrnecy * 16.61
        case "ZAR":
            outputCurrnecy.value = inputCurrnecy * 18.18
        case "AUD":
            outputCurrnecy.value = inputCurrnecy * 1.49
        case "NZD":
            outputCurrnecy.value = inputCurrnecy * 1.63
        case "CHF":
            outputCurrnecy.value = inputCurrnecy * 0.91

def CADCurrency(inputCurrnecy, outputCurrnecy, outputCurrnecyType):
    match outputCurrnecyType:
        case "USD":
            outputCurrnecy.value = inputCurrnecy * 0.73
        case "CAD":
            outputCurrnecy.value = inputCurrnecy * 1.0
        case "EUR":
            outputCurrnecy.value = inputCurrnecy * 0.68
        case "JMD":
            outputCurrnecy.value = inputCurrnecy * 114.0
        case "MXN":
            outputCurrnecy.value = inputCurrnecy * 12.20
        case "ZAR":
            outputCurrnecy.value = inputCurrnecy * 13.35
        case "AUD":
            outputCurrnecy.value = inputCurrnecy * 1.10
        case "NZD":
            outputCurrnecy.value = inputCurrnecy * 1.20
        case "CHF":
            outputCurrnecy.value = inputCurrnecy * 0.67

def EURCurrency(inputCurrnecy, outputCurrnecy, outputCurrnecyType):
    match outputCurrnecyType:
        case "USD":
            outputCurrnecy.value = inputCurrnecy * 1.09
        case "CAD":
            outputCurrnecy.value = inputCurrnecy * 1.48
        case "EUR":
            outputCurrnecy.value = inputCurrnecy * 1.0
        case "JMD":
            outputCurrnecy.value = inputCurrnecy * 169.06
        case "MXN":
            outputCurrnecy.value = inputCurrnecy * 18.03
        case "ZAR":
            outputCurrnecy.value = inputCurrnecy * 19.76
        case "AUD":
            outputCurrnecy.value = inputCurrnecy * 1.63
        case "NZD":
            outputCurrnecy.value = inputCurrnecy * 1.77
        case "CHF":
            outputCurrnecy.value = inputCurrnecy * 0.99

def JMDCurrency(inputCurrnecy, outputCurrnecy, outputCurrnecyType):
    match outputCurrnecyType:
        case "USD":
            outputCurrnecy.value = inputCurrnecy * 0.0064
        case "CAD":
            outputCurrnecy.value = inputCurrnecy * 0.0088
        case "EUR":
            outputCurrnecy.value = inputCurrnecy * 0.0059
        case "JMD":
            outputCurrnecy.value = inputCurrnecy * 1.0
        case "MXN":
            outputCurrnecy.value = inputCurrnecy * 0.11
        case "ZAR":
            outputCurrnecy.value = inputCurrnecy * 0.12
        case "AUD":
            outputCurrnecy.value = inputCurrnecy * 0.0096
        case "NZD":
            outputCurrnecy.value = inputCurrnecy * 0.011
        case "CHF":
            outputCurrnecy.value = inputCurrnecy * 0.0059

def MXNCurrency(inputCurrnecy, outputCurrnecy, outputCurrnecyType):
    match outputCurrnecyType:
        case "USD":
            outputCurrnecy.value = inputCurrnecy * 0.060
        case "CAD":
            outputCurrnecy.value = inputCurrnecy * 0.082
        case "EUR":
            outputCurrnecy.value = inputCurrnecy * 0.055
        case "JMD":
            outputCurrnecy.value = inputCurrnecy * 9.35
        case "MXN":
            outputCurrnecy.value = inputCurrnecy * 1.0
        case "ZAR":
            outputCurrnecy.value = inputCurrnecy * 1.09
        case "AUD":
            outputCurrnecy.value = inputCurrnecy * 0.090
        case "NZD":
            outputCurrnecy.value = inputCurrnecy * 0.098
        case "CHF":
            outputCurrnecy.value = inputCurrnecy * 0.055

def ZARCurrency(inputCurrnecy, outputCurrnecy, outputCurrnecyType):
    match outputCurrnecyType:
        case "USD":
            outputCurrnecy.value = inputCurrnecy * 0.055
        case "CAD":
            outputCurrnecy.value = inputCurrnecy * 0.075
        case "EUR":
            outputCurrnecy.value = inputCurrnecy * 0.051
        case "JMD":
            outputCurrnecy.value = inputCurrnecy * 8.54
        case "MXN":
            outputCurrnecy.value = inputCurrnecy * 0.91
        case "ZAR":
            outputCurrnecy.value = inputCurrnecy * 1.0
        case "AUD":
            outputCurrnecy.value = inputCurrnecy * 0.082
        case "NZD":
            outputCurrnecy.value = inputCurrnecy * 0.090
        case "CHF":
            outputCurrnecy.value = inputCurrnecy * 0.050

def AUDCurrency(inputCurrnecy, outputCurrnecy, outputCurrnecyType):
    match outputCurrnecyType:
        case "USD":
            outputCurrnecy.value = inputCurrnecy * 0.67
        case "CAD":
            outputCurrnecy.value = inputCurrnecy * 0.91
        case "EUR":
            outputCurrnecy.value = inputCurrnecy * 0.61
        case "JMD":
            outputCurrnecy.value = inputCurrnecy * 104.03
        case "MXN":
            outputCurrnecy.value = inputCurrnecy * 11.13
        case "ZAR":
            outputCurrnecy.value = inputCurrnecy * 12.17
        case "AUD":
            outputCurrnecy.value = inputCurrnecy * 1.0
        case "NZD":
            outputCurrnecy.value = inputCurrnecy * 1.09
        case "CHF":
            outputCurrnecy.value = inputCurrnecy * 0.61

def NZDCurrency(inputCurrnecy, outputCurrnecy, outputCurrnecyType):
    match outputCurrnecyType:
        case "USD":
            outputCurrnecy.value = inputCurrnecy * 0.61
        case "CAD":
            outputCurrnecy.value = inputCurrnecy * 0.83
        case "EUR":
            outputCurrnecy.value = inputCurrnecy * 0.56
        case "JMD":
            outputCurrnecy.value = inputCurrnecy * 95.20
        case "MXN":
            outputCurrnecy.value = inputCurrnecy * 10.18
        case "ZAR":
            outputCurrnecy.value = inputCurrnecy * 11.15
        case "AUD":
            outputCurrnecy.value = inputCurrnecy * 0.92
        case "NZD":
            outputCurrnecy.value = inputCurrnecy * 1.0
        case "CHF":
            outputCurrnecy.value = inputCurrnecy * 0.56

def CHFCurrency(inputCurrnecy, outputCurrnecy, outputCurrnecyType):
    match outputCurrnecyType:
        case "USD":
            outputCurrnecy.value = inputCurrnecy * 1.10
        case "CAD":
            outputCurrnecy.value = inputCurrnecy * 1.50
        case "EUR":
            outputCurrnecy.value = inputCurrnecy * 1.01
        case "JMD":
            outputCurrnecy.value = inputCurrnecy * 170.77
        case "MXN":
            outputCurrnecy.value = inputCurrnecy * 18.27
        case "ZAR":
            outputCurrnecy.value = inputCurrnecy * 20.00
        case "AUD":
            outputCurrnecy.value = inputCurrnecy * 1.64
        case "NZD":
            outputCurrnecy.value = inputCurrnecy * 1.79
        case "CHF":
            outputCurrnecy.value = inputCurrnecy * 1.0